content/named-tuples.md

```diff
@@ -0,0 +1,777 @@
+---
+layout: sip
+permalink: /sips/named-tuples.html
+stage: implementation
+status: waiting-for-implementation
+presip-thread: https://contributors.scala-lang.org/t/pre-sip-named-tuples/6403/164
+title: SIP-58 - Named Tuples
+---
+
+**By: Martin Odersky**
+
+## History
+
+| Date          | Version            |
+|---------------|--------------------|
+| Jan 13th 2024 | Initial Draft      |
+
+## Summary
+
+We propose to add new form of tuples where the elements are named.
+Named tuples can be types, terms, or patterns. Syntax examples:
+```scala
+type Person = (name: String, age: Int)
```

Comment:
What happens if someone writes the following:

```scala
type Person = (name: String, name: String)
```

Is this a syntax error, type error, or something else? The runtime implementation of `(String, String)` doesn't care about names, but then when someone calls `.name` the compiler cannot know which tuple entry it is referring to

Reply:
It's a type error. It's listed in the Restrictions section of the SIP.